fractal_server/main.py

```diff
@@ -140,6 +144,32 @@ def start_application() -> FastAPI:
             The fully initialised application.
     """
     app = FastAPI(lifespan=lifespan)
+
+    settings = Inject(get_settings)
+
+    @app.middleware("http")
```

Comment:
I suggest defining the middleware outside the `start_application` function, and then including it either as an attribute of `FastAPI(...)` or through `app.add_middleware`. This should make it easier to unit-test it (and maybe make the structure more readable - but that's just my opinion).